### PR TITLE
Upgrade to Node 16 and Serverless v3

### PR DESCRIPTION
**Pull request by mojotalantikite**
## Summary

This upgrades our environment to use Node 16. In order to get there we also moved to Serverless v3.x (from v2.x).

#### Related issues
https://qmacbis.atlassian.net/browse/MR-2277

#### Screenshots
All of our lambdas are running in the 16.x runtime:

Screen Shot 2022-06-23 at 1 50 11 PM

### CODE REVIEW

File: services/app-api/serverless.yml [hunk 1]
```diff
@@ -20,24 +19,22 @@ plugins:
...
   emailRateHelpAddress: ${env:SES_RATE_HELP_EMAIL_ADDRESS}
   emailDevTeamHelpAddress: ${env:SES_DEV_TEAM_HELP_EMAIL_ADDRESS}
-  auroraArn: ${env:CF_CONFIG_IGNORED_LOCALLY, cf:postgres-${self:custom.stage}.PostgresAuroraArn }
+  auroraArn: ${env:CF_CONFIG_IGNORED_LOCALLY, cf:postgres-${sls:stage}.PostgresAuroraArn }
   iamPath: ${env:IAM_PATH, "/"}
   iamPermissionsBoundaryPolicy: ${env:FULL_IAM_PERMISSIONS_BOUNDARY_POLICY, ""}
-  document_uploads_bucket_arn: ${env:CF_CONFIG_IGNORED_LOCALLY, cf:uploads-${self:custom.stage}.DocumentUploadsBucketArn}
-  secretsManagerSecret: aurora_postgres_${self:custom.stage} #pragma: allowlist secret
+  document_uploads_bucket_arn: ${env:CF_CONFIG_IGNORED_LOCALLY, cf:uploads-${sls:stage}.DocumentUploadsBucketArn}
+  secretsManagerSecret: aurora_postgres_${sls:stage} #pragma: allowlist secret
```
Comment (macrael) [inline services/app-api/serverless.yml:37]: what is the “sis” directive here?

**Review by macrael — APPROVED** [review]
Amazing this just took serverless.yml changes. Have you run `dev local` yet? I’m pulling this down now to see if the output has changed significantly.

File: services/app-api/serverless.yml (hunk 1 above)
Comment (mojotalantikite) [reply services/app-api/serverless.yml:37]: `sls` is just how serverless talks about itself. There's an `sls` command that is just a symlink to `serverless` as well for CLI commands. Not sure why they decided to do that.

**Comment by mojotalantikite** [comment]
Ok, seems like I've got all the random bits of serverless changes fixed for `./dev local`. There's just one error left from `useLocalStorage.ts`: 

```
Unable to set local storage. Error message: SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data
```

I'll start looking at that one after the weekend.

**Pull request merged**